src/connection.ts

```diff
@@ -42,9 +45,9 @@ export function createConnection(): lsp.Connection {
                 implementationProvider: true,
                 completionProvider: {
                     /**
-                     * only invoke completion once `.` is pressed
+                     * only invoke completion once `.` or `[` are pressed
                      */
-                    triggerCharacters: ['.'],
+                    triggerCharacters: COMPLETION_TRIGGERS.slice(),
```

Comment:
nit: I don't think we need to clone an array here

Reply:
This is the result of declaring this array as a `const`. I went ahead and removed it.